### DISCUSSION TITLE
Is there a way to make diamond return a higher identity subsequence/Diamond significantly undertrims candidate sequence.

### DISCUSSION THREAD

**Discussion by kevinmoran1988** [bbuchfink/diamond#691] [Q&A] [answered]
Sometimes Diamond correctly finds a match, but extends the match **far** past where it should end.

Using the following command.

> diamond blastx -d {diamond_db_path} -q {input_file.name} -o {out_path} --very-sensitive --masking 0 -e {args.evalue} --outfmt 6 qseqid sseqid qframe evalue bitscore qstart qend sstart send pident {quiet} --top {top_amount} --min-orf 20 --max-hsps 0 -p {num_threads}"

Example below. Diamond returns a match against these references for this sequence.

NODE_451470	28588:002869	-1	2.72e-23	88.6	287	66	3	81	64.6
NODE_451470	7227:002172	-1	2.72e-23	88.6	287	66	3	81	64.6
NODE_451470	7159:0010bd	-1	6.23e-22	85.1	287	66	3	82	63.8
NODE_451470	7165:0005df	-1	1.95e-20	81.3	287	66	3	82	58.8

You may notice that Pident is rather low.

What has happened is that Diamond has extended the correctly mapped section of our candidate sequence to include a poorly scoring segment that does not map.

In the below picture I've aligned the output coords of NODE_451470 to the references. NODE_451470 is the 1st sequence. You will see at the end there is a section that is pure junk.

![2023-03-20 (4)](https://user-images.githubusercontent.com/39282665/226462732-62d6ecce-e569-4493-b499-bedb03c7292e.png)

This is quite a frustrating problem, so I appreciate any advice you have to give.

Edit: I am using Diamond v2.0.14.152

**Comment by kevinmoran1988** [comment, marked as answer]
Using 2.1.6 returns a different (better) answer. 

NODE_451470	7227:002172	-1	1.08e-22	87.0	287	141	3	54	84.6
NODE_451470	28588:002869	-1	3.05e-22	85.9	287	141	3	54	82.7
NODE_451470	7159:0010bd	-1	6.96e-21	82.4	287	141	3	55	81.1
NODE_451470	7165:0005df	-1	7.74e-20	79.7	287	141	3	55	75.5

What version was this changed?  I go back through versions but it will take some time.

**Reply by bbuchfink** [reply]
I'm not sure what changed between the versions. If you send me the sequences I can look into it.
